lib/matplotlib/axes/_axes.py

```diff
@@ -2889,9 +2889,11 @@ def xywhere(xs, ys, mask):
                 # Check if xerr is scalar or symmetric. Asymmetric is handled
                 # above. This prevents Nx2 arrays from accidentally
                 # being accepted, when the user meant the 2xN transpose.
-                if not (len(xerr) == 1 or
-                        (len(xerr) == len(x) and not (
-                            iterable(xerr[0]) and len(xerr[0]) > 1))):
+                # special case for empty lists
+                if len(xerr) and not (len(xerr) == 1 or
```

Comment:
If I'm not mistaken, these two conditions can be combined into `if len(xerr) > 1 and not (everything on the next two lines):`.